packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleImageViewer/SingleImageViewerContainer.js

```diff
@@ -36,29 +40,41 @@ class SingleImageViewerContainer extends React.Component {
   async handleSubject () {
     const { subject } = this.props
     // TODO: Add polyfill for Object.values for IE
-    const imageUrl = Object.values(subject.locations[0])[0]
     this.setState({ loading: asyncStates.loading })
     try {
+      const imageUrl = Object.values(subject.locations[0])[0]
       const img = await this.fetchImage(imageUrl)
+      const svg = this.imageViewer.current
 
       this.setState({
-        height: img.height,
-        width: img.width,
+        clientHeight: svg.clientHeight,
+        clientWidth: svg.clientWidth,
+        naturalHeight: img.naturalHeight,
+        naturalWidth: img.naturalWidth,
         loading: asyncStates.success
       })
     } catch (error) {
       this.onError(error)
     }
   }
 
+  onLoad (event) {
+    const { onReady } = this.props
+    const { clientHeight, clientWidth, naturalHeight, naturalWidth } = this.state
+    const { target } = event || {}
```

Comment:
When would the target be undefined?

Reply:
Video and audio, probably, from what I remember of the `onLoad` callbacks in PFE.

Reply:
This line catches the case where the event is undefined, though.

Reply:
Oh wait, this is the image loading code. Ignore my comments, I was thinking of the generic event handler in the subject viewer store.